src/components/SkeletonLoader.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react'
+import { StyleSheet } from 'react-native'
+import MaskedView from '@react-native-masked-view/masked-view'
+import {
+  useSharedValue,
+  withRepeat,
+  withTiming,
+  useAnimatedStyle,
+  interpolate,
+} from 'react-native-reanimated'
+import { Box, theme } from 'utils/theme'
+import LinearGradient from 'react-native-linear-gradient'
+import { AnimatedBox } from './AnimatedBox'
+
+type Props = {
+  children: React.ReactElement
+  width: number
+  isLoading: boolean
+}
+
+export const SkeletonLoader = (props: Props) => {
+  const loaderValue = useSharedValue(0)
+
+  useEffect(() => {
+    loaderValue.value = withRepeat(withTiming(1, { duration: 1500 }), Infinity)
+
+    return () => {
+      loaderValue.value = 0
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [
+      {
+        translateX: interpolate(loaderValue.value, [0, 1], [-props.width, props.width]),
+      },
+    ],
+  }))
```

Comment:
Can we move this to a hook, e.g. `const { animatedStyle } = useSkeletonLoaderStyle()`? This hook doesn't need to be moved out of this file, as it probably won't be reused, but still I think it'd be easier to tweak just the animation or just the gradient appearance with this approach. Also why exhaustive-deps is not needed there? Could you please add a comment with some context to the code?

Reply:
Dependency array check is blocked here because we want to run this only on initial render of the screen. I will add the comment there